$OpenBSD: patch-src_sg_h_in,v 1.2 2013/08/12 20:01:49 jeremy Exp $

squidGuard 1.4 patch 20091019; increase MAX_BUF above squid's MAX_URL value.

Fix segfault on 64-bit due to calling an undeclared function.  The function
is supposed to return a pointer (64-bit) but when undeclared returns an int
(32-bit). 

--- src/sg.h.in.orig	Fri Nov 16 08:58:32 2007
+++ src/sg.h.in	Mon Aug 12 02:40:28 2013
@@ -73,7 +73,7 @@ int tolower();
 #define REQUEST_TYPE_REDIRECT   2
 #define REQUEST_TYPE_PASS       3
 
-#define MAX_BUF 4096
+#define MAX_BUF 12288
 
 #define DEFAULT_LOGFILE "squidGuard.log"
 #define WARNING_LOGFILE "squidGuard.log"
@@ -412,6 +412,7 @@ char   *niso __P((time_t));
 struct UserQuotaInfo *setuserquota __P(());
 void sgSourceUserQuota __P((char *, char *, char *));
 
+struct UserInfo *setuserinfo();
 
 void   *sgMalloc __P((size_t));
 void   *sgCalloc __P((size_t, size_t));
